python/Ganga/GPIDev/Lib/File/OutputFileManager.py

```diff
@@ -210,53 +195,38 @@ def getWNCodeForDownloadingInputFiles(job, indent):
     """
 
     from Ganga.GPIDev.Lib.Dataset.GangaDataset import GangaDataset
-    if job.inputfiles is None or len(job.inputfiles) == 0 and\
-            (not job.inputdata or ((not isType(job.inputdata, GangaDataset)) or\
-                not job.inputdata.treat_as_inputfiles )):
-        return ""
+    if job.inputfiles is None or len(job.inputfiles) == 0 and not (job.inputdata or (isinstance(job.inputdata, GangaDataset) and job.inputdata.treat_as_inputfiles)):
+        if job.master is None or len(job.master.inputfiles) == 0 and not (job.master.inputdata or (isinstance(job.master.inputdata, GangaDataset) and job.master.inputdata.treat_as_inputfiles)):
```

Comment:
These two lines are very hard to parse, I have no idea what they are doing. Is there any way we could simplify them?